lib/validators.rb

```diff
@@ -137,9 +137,107 @@ def check_shipments_specificities
       }
     end
 
+    def vehicle_days_availability(vehicles_set, timewindows_set)
```

Comment:
I think we can be more explicit here, for example `calculate_day_availabilities(vehicles, timewindows)` is more explanatory and gives the raison d'etre

We don't need the `set`s beucase they are not sets but also just `vehicles` and `timewindows` are enough.